### PR TITLE
Enhancement/fake app

### LINKED ISSUE #12: Add Fake App
Add [app](https://github.com/faker-ruby/faker/blob/master/doc/default/app.md), which includes `name`, `version`, `author`, and `semanticVersion`. This should also include options to change the version's major/minor/patch values as well as being able to say if a library is early release or stable.

### PR DESCRIPTION
**Pull request by Caps-Looking**
Closes #12 

## Description
Creation of a wrapper of fake apps.

## Technical Details
- Changed `FakeContext` to add `fakeApp`.
- Added `FakeApp` class to provide the `fakeApp`.
- Changed `FakerTypeAliases` to add `FakerApp` alias.
- Changed `FakerContextTest` to add `fakeApp` test.
- Added `FakeAppTest` to test `FakeApp`.
- Added `SemanticVersion` class to generate the app semantic version.

## Code Samples
Please provide code samples for how these changes will be used by applications consuming this library. Here's an example:

```kotlin
fun main() {
    val semanticVersion = SemanticVersion()
    semanticVersion.changePatch(2)
    val fakeApp: FakeApp(name="Treeflex", version="1.85", author="Daphne Swift", semanticVersion=semanticVersion)
    fakek { println("App: Name - ${fakeApp.name}, Version - ${fakeApp.version}, Author - ${fakeApp.author}, Semantic Version - ${fakeApp.semanticVersion}") }
}
```

IDK if that's what you thought about architecture to create the semantic version, or if these are the right parameters.
Let me know what you think about it and your suggestions.

Reviewers: @CodyEngel

### CODE REVIEW

File: src/main/kotlin/dev/fakek/fakers/SemanticVersion.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package dev.fakek.fakers
```
Comment (CodyEngel) [inline src/main/kotlin/dev/fakek/fakers/SemanticVersion.kt:1]: Thoughts on moving this to `dev.fakek.fakes`?

File: src/main/kotlin/dev/fakek/fakers/SemanticVersion.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import kotlin.random.Random
+
+/**
+ * SemanticVersion provides a semantic version which includes major, minor, and patch.
+ *
+ * @param major is the major number.
+ * @param minor is the minor number.
+ * @param patch is the patch number.
+ */
+class SemanticVersion(private var major: Int, private var minor: Int, private var patch: Int) {
```
Comment (CodyEngel) [inline src/main/kotlin/dev/fakek/fakers/SemanticVersion.kt:12]: Thoughts on renaming to `FakeSemanticVersion`?

**Review by CodyEngel — COMMENTED** [review]
Hey @Caps-Looking had a couple of questions around `SemanticVersion`, I'm wondering if it'd make sense to include this as it's own `Fake`, but the changes are looking good.

**Comment by CodyEngel** [comment]
Hey @Caps-Looking -- tests are currently failing on CI. Once those are resolved I can go ahead and review & merge the changes.

**Comment by Caps-Looking** [comment]
Hi @CodyEngel!

Sorry for the delay, got covid and my life is been crazy since then.
I just merged with main, and it seems everything is working now.

**Comment by CodyEngel** [comment]
Whoa @Caps-Looking welcome back! Sorry about my late response back, life has been hectic at work. Really sorry to hear that you got covid but I'm happy to see you are doing better. I'll go ahead and get this merged in 💪

**Review by CodyEngel — APPROVED** [review]
LGTM :shipit:

**Pull request merged**